transports/jms/src/test/java/org/mule/transport/jms/JmsConnectorTestCase.java

```diff
@@ -199,9 +206,51 @@ public void createConnectionFactoryWrapperWhenUsingTransactionManager() throws E
         JmsConnector connector = createConnectionFactoryWhenGettingConnection(mock(TestXAConnectionFactory.class));
         assertThat(connector.getConnectionFactory(), instanceOf(DefaultXAConnectionFactoryWrapper.class));
     }
+    
+    @Test
+    public void doesNotDirtyNotificationThreadClassLoaderOnNotification() throws Exception
```

Comment:
This tests passes with any changes in the production code. I think we need two tests: 
1) Test checking that the notification is call using the app classloader
2) Test checking that classloader is restored after notification processing